modules/swagger-codegen/src/main/resources/aspnetcore/controller.mustache

```diff
@@ -31,9 +31,10 @@ namespace {{packageName}}.Controllers
         [{{httpMethod}}]
         [Route("{{{basePathWithoutHost}}}{{{path}}}")]
         [ValidateModelState]
-        [SwaggerOperation("{{operationId}}")]{{#responses}}{{#returnType}}
-        [SwaggerResponse({{code}}, typeof({{&returnType}}), "{{message}}")]{{/returnType}}{{/responses}}
-        public virtual {{#returnType}}IActionResult{{/returnType}}{{^returnType}}void{{/returnType}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
+        [SwaggerOperation("{{operationId}}")]{{#responses}}{{#dataType}}
+        [SwaggerResponse(statusCode: {{code}}, type: typeof({{&dataType}}), description: "{{message}}")]{{/dataType}}{{^dataType}}
+        [SwaggerResponse(statusCode: {{code}}, description: "{{message}}")]{{/dataType}}{{/responses}}
+        public virtual IActionResult {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
```

Comment:
It's not really intuitive here, but swagger specs documented without a return type will cause a compilation error if you specify `IActionResult` rather than `void`. At least it did when I wrote the template. Can you verify with a spec that has responses with void (undefined) return types.

Reply:
Not sure I get your comment right..?

Surely you have to return *something*, or throw an exception..

E.g.
```
        [HttpGet]
        public IActionResult Get()
        {
            return NoContent(); // 200
        }
        [HttpGet]
        public IActionResult Get2()
        {
            throw new NotImplementedException();
        }
```

Having VOID as return type limits people from returning ActionResults (which map to Statuscodes).. Working with beginners a lot that confused them (how do they return anything if the return value is void..)

Let me know if this is what you mean.. :) Thanks!